src/engine/specification.md

```diff
@@ -219,8 +219,11 @@ Payload validation process consists of validating a payload with respect to the
 
 3. Client software **MUST** validate a payload according to the block header and execution environment rule set with modifications to these rule sets defined in the [Block Validity](https://eips.ethereum.org/EIPS/eip-3675#block-validity) section of [EIP-3675](https://eips.ethereum.org/EIPS/eip-3675#specification):
   * If validation succeeds, the response **MUST** contain `{status: VALID, latestValidHash: payload.blockHash}`
-  * If validation fails, the response **MUST** contain `{status: INVALID, latestValidHash: validHash}` where `validHash` is the block hash of the most recent *valid* ancestor of the invalid payload. That is, the valid ancestor of the payload with the highest `blockNumber`
-  * If the most recent valid ancestor is a PoW block, `latestValidHash` **MUST** be set to `0x0000000000000000000000000000000000000000000000000000000000000000`
+  * If validation fails, the response **MUST** contain `{status: INVALID, latestValidHash: validHash}` where `validHash` is the block hash of the unique ancestor of the invalid payload satisfying the following two conditions:
+    - It is fully validated and deemed `VALID`
+    - Any other ancestor of the invalid payload with a higher `blockNumber` is `INVALID`
+
+    if such a block exists and is a PoS block. `validHash` is `0x0000000000000000000000000000000000000000000000000000000000000000` if such a block exists and it is a PoW block, and `null` otherwise. 
```

Comment:
```suggestion
    if such block is a PoS block. `validHash` is `0x0000000000000000000000000000000000000000000000000000000000000000` if such block is a PoW block. If the Execution Layer cannot determine such block precisely, it may return `null` instead of the block hash.
```

Reply:
What about something like the following:

* If validation fails, the response **MUST** contain `{status: INVALID, latestValidHash: validHash}` where `validHash` **MUST** be:
	* The block hash of ancestor of the invalid payload if the following two conditions are satisfied:
	  * it is fully validated and deemed **VALID**
	  * any other ancestor of the invalid payload with a higher `blockNumber` is `INVALID`
	* `0x0000000000000000000000000000000000000000000000000000000000000000` if the above conditions are satisfied by a PoW block
	* `null` if client software can't determine ancestor of the invalid payload satisfying the above conditions

Reply:
that looks good to me

Reply:
pushed this, please let me know if this doesn't work.